apps/token/src/components/transactions-modal/transactions-modal.tsx

```diff
@@ -65,53 +48,52 @@ export const TransactionModal = () => {
 
     return (
       <TransactionModalStatus>
-        <Loader invert={true} />
-        <span>
-          {t('confirmationsRemaining', {
-            confirmations: txObj.receipt.confirmations,
-            required: txObj.requiredConfirmations,
-          })}
-        </span>
+        {t('confirmationsRemaining', {
+          confirmations: txObj.receipt.confirmations,
+          required: txObj.requiredConfirmations,
+        })}
       </TransactionModalStatus>
     );
   };
 
   return (
-    <Overlay
-      className="bp3-dark"
-      isOpen={appState.transactionOverlay}
-      onClose={close}
-      transitionDuration={0}
+    <Dialog
+      open={appState.transactionOverlay}
+      title={t('ethTransactionModalTitle')}
+      onChange={(isOpen) =>
+        appDispatch({
+          type: AppStateActionType.SET_TRANSACTION_OVERLAY,
+          isOpen,
+        })
+      }
     >
-      <Modal title={t('ethTransactionModalTitle')}>
-        {transactions.length ? (
-          <table className="w-full">
-            <thead>
-              <tr>
-                <TransactionModalTh>{t('transaction')}</TransactionModalTh>
-                <TransactionModalTh>{t('status')}</TransactionModalTh>
-              </tr>
-            </thead>
-            <tbody>
-              {transactions.map((t) => {
-                return (
-                  <tr key={t.tx.hash}>
-                    <TransactionModalTd>
-                      <EtherscanLink
-                        tx={t.tx.hash}
-                        text={truncateMiddle(t.tx.hash)}
-                      />
-                    </TransactionModalTd>
-                    <TransactionModalTd>{renderStatus(t)}</TransactionModalTd>
-                  </tr>
-                );
-              })}
-            </tbody>
-          </table>
-        ) : (
-          <p>{t('noTransactions')}</p>
-        )}
-      </Modal>
-    </Overlay>
+      {transactions.length ? (
+        <table className="w-full">
+          <thead>
+            <tr>
+              <TransactionModalTh>{t('transaction')}</TransactionModalTh>
+              <TransactionModalTh>{t('status')}</TransactionModalTh>
+            </tr>
+          </thead>
+          <tbody>
+            {transactions.map((t) => {
+              return (
```

Comment:
Suggestion (non-blocking): could do implicit return here too (clean, reduces lines of code, easy to read, win win win IMO)